### PR TITLE
Added short Gulp Section to Intro

### PR DESCRIPTION
**Pull request by zeldar**
Issue #743 - Adding Gulp.  
The basics have been added, but I will continue to add onto this throughout the week for another commit in the short future.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -38,6 +38,16 @@ $ npm install -g grunt-cli
 $ gem install sass
 ```
 
+```bash
+$ npm install -g grunt-cli
+```
+
+* Gulp - You may use Gulp for Live Reload, Linting, and SASS or LESS.
```
Comment (lirantal) [inline README.md]: We should make sure that it is explicitly says that Gulp right now is optional, not mandatory.

File: README.md [hunk 2]
```diff
@@ -152,6 +162,64 @@ And to run only the client tests, run the test:client task:
...
+```
+or
+
+```bash
+$ gulp default
+```
+
+The server is now running on http://localhost:3000 if you are using the default settings. 
+
+### Errors at Runtime
```
Comment (lirantal) [inline README.md]: I don't think it's a good idea to have a troubleshooting section in the README.
We should have a troubleshooting section in the documentation and point to it from the README

File: README.md [hunk 3]
```diff
@@ -152,6 +162,64 @@ And to run only the client tests, run the test:client task:
...
+
+### Running in Production mode
+To run your application with *production* environment configuration, execute gulp as follows:
+
+```bash
+$ gulp prod
+```
+
+### Testing Your Application with Gulp
+You can run the full test suite included with MEAN.JS with the test task:
```
Comment (lirantal) [inline README.md]: s/with/using

File: README.md (hunk 1 above)
Comment (zeldar) [inline README.md]: Agreed.  I will make that more clear.

File: README.md (hunk 2 above)
Comment (zeldar) [inline README.md]: I will remove that section.   Will look into a/the troubleshooting guide.  I apologize, still catching the ropes of contribution.  I'll make the changes and recommit after work today.  I'm hoping to make it easier for people to get started with MEAN.js and those were some of the roadblocks I encountered.

Comment (lirantal) [inline README.md]: No problems, good work.

File: README.md [hunk 4]
```diff
@@ -152,6 +162,64 @@ And to run only the client tests, run the test:client task:
...
+
+```bash
+$ gulp test
+```
```
Comment (jloveland) [inline README.md]: do we also want to add the following to indicate you can test each tier?
```bash
# run server tests
gulp test:server

# run client tests
gulp test:client

# run e2e tests
gulp test:e2e
```

**Comment by codydaig** [comment]
@zeldar Thanks for the contribution! If you can adjust the line comments above, and move the troubleshooting out of this PR and into a PR against the gh-pages (documentation) branch, it should be good to go!

File: README.md (hunk 4 above)
Comment (zeldar) [inline README.md]: @jloveland I like that idea.  I will include your addition.

**Comment by ilanbiala** [comment]
@codydaig now we just have to move the troubleshooting out of this PR right?

**Comment by codydaig** [comment]
@ilanbiala @zeldar Correct me if I am wrong, but looks like the troubleshooting got removed. If so I think it's ready.

**Comment by ilanbiala** [comment]
Yep, but it still needs to be added to gh-pages in a separate PR. Let's get this squashed and merged?

**Comment by lirantal** [comment]
This shouldn't hold 0.4.2 so I set it to 0.5.0

**Comment by ilanbiala** [comment]
Just to clarify: @zeldar we'll still merge this whenever it comes around, so if it comes before 0.4.2 releases, great. If not, that's fine too, but we still want to merge this in.

**Comment by zeldar** [comment]
@ilanbiala Yes, Troubleshooting was removed. 

Do you want the addition of the Troubleshooting section pushed to this commit before squashing, or do you want me to create a new issue for Troubleshooting and push the Gulp Troubleshooting as a different commit?  Just wanted to clarify, but can have that finished this evening.

**Comment by codydaig** [comment]
@zeldar The troubleshooting has to be to a separate PR since it will be against the gh-pages branch to have on the website / documentation.

**Comment by zeldar** [comment]
@codydaig #1037 PR for gh-pages Gulp documentation and troubleshooting.  

I have the same thing I wrote into the HTML in a TROUBLESHOOTING.md as well on my local machine, but noticed the TROUBLESHOOTING.md was blank.  Is there plans for that document, or is it part of the Jekyll process?  

Do you need anything else from me on this PR?  Thank you for the help in getting sorted out.

**Comment by codydaig** [comment]
@zeldar There aren't any plans for a TROUBLESHOOTING.md document.

**Comment by codydaig** [comment]
LGTM

**Comment by ilanbiala** [comment]
@zeldar @codydaig nice, looks like we can put this in for 0.4.2. @codydaig do you want to already start reformatting commit messages?

**Comment by codydaig** [comment]
@ilanbiala We can't realistically start enforcing it until the changelog.md doc gets merged in. I'm starting too on my commits but I don't think we should enforce until 0.4.2 is released. All commits for 0.5.0 should follow the format.

**Comment by ilanbiala** [comment]
Yeah makes sense.

**Comment by codydaig** [comment]
@zeldar Can you squash down to 1 commit? Ping me once it's squashed and I'll merge.

**Comment by zeldar** [comment]
@codydaig I believe this is ready. Thank you!

**Comment by codydaig** [comment]
@zeldar Yup! Thanks a bunch! 

LGTM

**Pull request merged**